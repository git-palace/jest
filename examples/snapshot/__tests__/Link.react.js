// Copyright 2004-present Facebook. All Rights Reserved.
/* eslint-disable no-unused-vars */

'use strict';

const React = require('react');
const Link = require('../Link.react');
const renderer = require('react/lib/ReactTestRenderer');

describe('Link', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <Link page="http://www.example.com">HI</Link>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders as an anchor when no page is set', () => {
    const tree = renderer.create(
      <Link>HI</Link>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('changes the class when hovered', () => {
    const component = renderer.create(
      <Link page="http://www.example.com">HI</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });

});
