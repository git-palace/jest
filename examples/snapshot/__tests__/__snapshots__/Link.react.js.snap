exports[`Link renders correctly 0`] = `
<a
  className="normal"
  href="http://www.example.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  HI
</a>
`;

exports[`Link renders as an anchor when no page is set 0`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  HI
</a>
`;

exports[`Link changes the class when hovered 0`] = `
<a
  className="normal"
  href="http://www.example.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  HI
</a>
`;

exports[`Link changes the class when hovered 1`] = `
<a
  className="hovered"
  href="http://www.example.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  HI
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="normal"
  href="http://www.example.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  HI
</a>
`;
